File: HRMS/HRMS.Web/Modules/ServerTypes/HR/DepartmentRow.ts
import { fieldsProxy, getLookup, getLookupAsync } from "@serenity-is/corelib";

export interface DepartmentRow {
    DepartmentId?: number;
    DepartmentName?: string;
    Description?: string;
    IsActive?: boolean;
}

export abstract class DepartmentRow {
    static readonly idProperty = 'DepartmentId';
    static readonly nameProperty = 'DepartmentName';
    static readonly localTextPrefix = 'HR.Department';
    static readonly lookupKey = 'HR.Department';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<DepartmentRow>('HR.Department') }
    static async getLookupAsync() { return getLookupAsync<DepartmentRow>('HR.Department') }

    static readonly deletePermission = 'HR:Department';
    static readonly insertPermission = 'HR:Department';
    static readonly readPermission = 'HR:Department';
    static readonly updatePermission = 'HR:Department';

    static readonly Fields = fieldsProxy<DepartmentRow>();
}

File: HRMS/HRMS.Web/Modules/HR/Department/DepartmentRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace HRMS.HR;

[ConnectionKey("Default"), Module("HR"), TableName("Departments")]
[DisplayName("Departments"), InstanceName("Department")]
[ReadPermission("HR:Department")]
[ModifyPermission("HR:Department")]
[LookupScript]
public sealed class DepartmentRow : Row<DepartmentRow.RowFields>, IIdRow, INameRow
{
    [DisplayName("Department Id"), Identity, IdProperty]
    public int? DepartmentId
    {
        get => fields.DepartmentId[this];
        set => fields.DepartmentId[this] = value;
    }

    [DisplayName("Department Name"), Size(100), NotNull, QuickSearch, NameProperty]
    public string DepartmentName
    {
        get => fields.DepartmentName[this];
        set => fields.DepartmentName[this] = value;
    }

    [DisplayName("Description")]
    public string Description
    {
        get => fields.Description[this];
        set => fields.Description[this] = value;
    }

    [DisplayName("Is Active"), NotNull]
    public bool? IsActive
    {
        get => fields.IsActive[this];
        set => fields.IsActive[this] = value;
    }

    public class RowFields : RowFieldsBase
    {
        public Int32Field DepartmentId;
        public StringField DepartmentName;
        public StringField Description;
        public BooleanField IsActive;
    }
}
